test/unit/mdc-drawer/mdc-drawer.test.js

```diff
@@ -104,35 +73,36 @@ test('#set open false calls foundation.close', () => {
 });
 
 test('keydown event calls foundation.handleKeydown method', () => {
-  const {component} = setupTest();
+  const {component, drawer} = setupTest();
   component.foundation_.handleKeydown = td.func();
-  component.initialSyncWithDOM();
-  domEvents.emit(document, 'keydown');
+  drawer.querySelector('.demo-drawer-list-item').focus();
+  domEvents.emit(drawer, 'keydown');
   td.verify(component.foundation_.handleKeydown(td.matchers.isA(Object)), {times: 1});
 });
 
 test('transitionend event calls foundation.handleTransitionEnd method', () => {
   const {component, drawer} = setupTest();
   component.foundation_.handleTransitionEnd = td.func();
-  component.initialSyncWithDOM();
   domEvents.emit(drawer, 'transitionend');
   td.verify(component.foundation_.handleTransitionEnd(td.matchers.isA(Object)), {times: 1});
 });
 
 test('#destroy removes keydown event listener', () => {
-  const {component} = setupTest();
-  const originalRemove = document.removeEventListener;
-  document.removeEventListener = td.func();
+  const {component, drawer} = setupTest();
+  const originalRemove = drawer.removeEventListener;
```

Comment:
There's probably a more natural way of performing these event handler tests... see example in menu: https://github.com/material-components/material-components-web/blob/1245573ea4350682b6548af89a43171302624e63/test/unit/mdc-menu/mdc-menu.test.js#L126-L140